service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java

```diff
@@ -905,89 +905,104 @@ private void setValueForOrderStatusIsCancelledOrDoneAsTrue(OrderStatusTranslatio
      */
 
     @Override
-    public List<OrderDetailInfoDto> setOrderDetail(List<UpdateOrderDetailDto> request, String language, String uuid) {
-        OrderDetailDto dto = new OrderDetailDto();
-        for (UpdateOrderDetailDto updateOrderDetailDto : request) {
-            if (Boolean.FALSE.equals(!updateOrderRepository.ifRecordExist(updateOrderDetailDto.getOrderId(),
-                updateOrderDetailDto.getBagId().longValue()))) {
-                updateOrderRepository.insertNewRecord(updateOrderDetailDto.getOrderId(),
-                    updateOrderDetailDto.getBagId().longValue());
-            }
-            if (nonNull(updateOrderDetailDto.getAmount())) {
-                updateOrderRepository
-                    .updateAmount(updateOrderDetailDto.getAmount(), updateOrderDetailDto.getOrderId(),
-                        updateOrderDetailDto.getBagId().longValue());
-            }
-            if (nonNull(updateOrderDetailDto.getExportedQuantity())) {
+    public void setOrderDetail(Long orderId,
+        Map<Integer, Integer> confirmed, Map<Integer, Integer> exported, String language, String uuid) {
+        final User currentUser = userRepository.findUserByUuid(uuid)
+            .orElseThrow(() -> new UserNotFoundException(USER_WITH_CURRENT_ID_DOES_NOT_EXIST));
+        collectEventsAboutSetOrderDetails(confirmed, exported, orderId, currentUser, language);
+
+        if (nonNull(exported)) {
```

Comment:
blocks for `exported` and  `confirmed` seem to be the same and can be extracted into separate method.

Reply:
I can't do this cause we use different methods for  updating values inside this blocks
.updateExporter(...)  and  .updateConfirm(...)